lightning/src/ln/monitor_tests.rs

```diff
@@ -259,6 +259,21 @@ fn do_chanmon_claim_value_coop_close(anchors: bool) {
 
 	check_closed_event!(nodes[0], 1, ClosureReason::LocallyInitiatedCooperativeClosure, [nodes[1].node.get_our_node_id()], 1000000);
 	check_closed_event!(nodes[1], 1, ClosureReason::CounterpartyInitiatedCooperativeClosure, [nodes[0].node.get_our_node_id()], 1000000);
+
+	assert!(nodes[0].chain_monitor.chain_monitor.get_and_clear_pending_events().is_empty());
+	assert!(nodes[1].chain_monitor.chain_monitor.get_and_clear_pending_events().is_empty());
+
+	// Test we can archived fully resolved channel monitor.
+	assert_eq!(nodes[0].chain_monitor.chain_monitor.list_monitors().len(), 1); // one monitor
+	// first archive should set balances_empty_height to current block height
+	nodes[0].chain_monitor.chain_monitor.archive_fully_resolved_channel_monitors(vec![funding_outpoint]); 
+	connect_blocks(&nodes[0], 2016);
+	// Second call after 2016 blocks, should archive the monitor
+	nodes[0].chain_monitor.chain_monitor.archive_fully_resolved_channel_monitors(vec![funding_outpoint]);
+	assert_eq!(nodes[0].chain_monitor.chain_monitor.list_monitors().len(), 0); // no monitor
+	// cleanup
+	nodes.get_mut(0).unwrap().chain_source.remove_watched_txn_and_outputs(funding_outpoint,
```

Comment:
@TheBlueMatt any thoughts about this? why do you think this is needed? feels a bit hacky to me

I added it because the test would fail in the `drop` phase asserting the data so i remove it here in `removed_watched_txn_outputs` to make it pass

Reply:
it would fail in this assertion https://github.com/lightningdevkit/rust-lightning/blob/main/lightning/src/ln/functional_test_utils.rs#L674